### PR TITLE
Add Markdownlint to docs

### PR DESCRIPTION
**Pull request by kagarmoe**
Signed-off-by: kagarmoe <kgarmoe@chef.io>

### Description

Adds [Markdownlint](https://github.com/DavidAnson/markdownlint) as a linter in chef-web-docs.
[markdownlint-cli2](https://www.npmjs.com/package/markdownlint-cli2)
A customized [.markdownlint.yaml](https://github.com/DavidAnson/markdownlint/blob/main/schema/.markdownlint.yaml)

To run: 
Add the [Markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) extension to VSCode
Changes to rules in the `.markdownlint.yaml` are applied immediately

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussion that's relevant]

### Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

### CODE REVIEW

**Comment by IanMadd** [comment]
Do we need markdownlint-cli2? It seems odd that it's installed with NPM. 
The config markdownlint.yaml is working with vscode. We could also install markdownlint-cli with brew.

**Comment by kagarmoe** [comment]
[Markdownlint-cli2](https://github.com/DavidAnson/markdownlint-cli2) is much faster & supports configuration files. I added it with NPM so it is automatically present for anyone that clones the repo.

**Comment by IanMadd** [comment]
I think this is just going to make markdownlint-cli2 available as a resource in chef-web-docs, but it's not going to be available in the CLI.

**Comment by IanMadd** [comment]
I worked on this a bit and I think the best way to go is for the docs team to install markdownlint-cli2 (or maybe just the first version depending on the features we want) using the global install `npm install markdownlint-cli2 --global` and get rid of the package install. I don't think the package install is doing anything for us. I haven't been able to figure out how to get the package install to return any errors by itself.

After doing the global install `markdownlint-cli2 content/**/*.md` returned a whole bunch of issues.

**Comment by IanMadd** [comment]
Also the duplicate package files and node_module dirs seems to be giving Hugo problems, deleting those made Hugo build properly.

**Comment by kagarmoe** [comment]
Ok, in this version you need to install Markdownlint-CLI2 locally. The other NPM files should be fine.

File: themes/docs-new/package-lock.json [hunk 1]
```diff
@@ -488,6 +488,57 @@
       "integrity": "sha1-FkpUg+Yw+kMh5a8HAg5TGDGyYJs=",
       "peer": true
     },
+    "node_modules/caniuse-api": {
```
Comment (IanMadd) [inline themes/docs-new/package-lock.json]: I think we don't need the changes in the package-lock file.

**Comment by IanMadd** [comment]
I also prefer having this run locally instead of with GH Actions because of the gigantic number of MD lint errors that we have.

**Comment by IanMadd** [comment]
Now my thinking is that we get this running locally so Docs Team members can hunt down MD errors locally for anything and then see if we can get MDLint with Reviewdog working for PR changes. https://github.com/marketplace/actions/run-markdownlint-with-reviewdog

**Comment by kagarmoe** [comment]
I agree @IanMadd. My plan was to fix the markdown first and then introduce it as a GH action.

**Review by IanMadd — APPROVED** [review]
LGTM 
tenor-78784022

**Pull request merged**